R/clustering.R

```diff
@@ -476,3 +480,42 @@ RunModularityClustering <- function(
   )
   return(clusters)
 }
+
+# Set up Leiden clustering algorithm
+# @name Leiden State Matrix
+# @rdname leiden
+#
+# @title Run Leiden clustering algorithm
+#
+# @description Implements the Leiden clustering algorithm in R using reticulate to run the Python version. Requires the python "leidenalg" and "igraph" modules to be installed. Returns a vector of partition indices.
+#
+# @param object A seurat class object.
+# @param ... Parameters to pass to the Python leidenalg function.
+#
+# @keywords graph network igraph mvtnorm simulation
+# @import reticulate
+# @importFrom igraph graph_from_adjacency_matrix write.graph
+# @export
+
+RunLeiden <- function(object, ...){
+  if (!py_module_available(module = 'leidenalg')) {
+    stop("Cannot find Leiden algorithm, please install through pip (e.g. pip install leidenalg).")
+  }
+  
+  #import python modules with reticulate
+  leidenalg <- reticulate::import("leidenalg")
+  ig <- reticulate::import("igraph")
+  
+  #export graph structure to call from Python
+  snn_graph <- object@snn
+  
+  #passes iGraph object to working directory to pass to python
+  igraph::write.graph(snn_graph, file = ".graph.edges", format = "graphml")
```

Comment:
Please use `tempfile` to make a temporary file in the system's temporary directory. Writing in locations that aren't a system's temporary directory violates [CRAN's policies](https://cran.r-project.org/web/packages/policies.html)

> - Packages should not write in the user’s home filespace (including clipboards), nor anywhere else on the file system apart from the R session’s temporary directory (or during installation in the location pointed to by TMPDIR: and such usage should be cleaned up). Installing into the system’s R installation (e.g., scripts to its bin directory) is not allowed.